### PR TITLE
Updates to tools for unblinding checks

### PR DESCRIPTION
**Pull request by kmohrman**
This PR includes most of the minor updates I've been making since we began unblinding, but none of these changes should actually impact the results (i.e. they do not change our analysis pkl files, or the datacards etc., they only provide updates to our tools for the studies we've been doing since unblinding). 

I don't think this PR is ready to be merged, as most of the changes are quite messy. However, I wanted to make the PR now, just so I can get an idea of what changes are included here, and clean things up when I get a chance, so that we can eventually get this into the master (since it's getting to the point where I'd like to have at least some of these updates in the master branch).

### CODE REVIEW

**Comment by kmohrman** [comment]
Note that the updates to the `topeft.py` processor and `selection.py` are probably not relevant enough to necessitate merging into the master (the updates were mostly for the additional variables and histograms we looked into during the unblinding cross checks), so I have restored these files to the master branch version.

The modifications have been moved to the `additional-histos` brand, documented on [this closed PR 325](https://github.com/TopEFT/topcoffea/pull/325)

**Comment by kmohrman** [comment]
I think this PR should now be ready to be merged (assuming CI passes). The updates are mainly to the tools we use to study the output pkl files (the plotter script, and the yield getting scripts), so this PR should have no impact whatsoever on the actual content of the output pkl file.

File: analysis/topEFT/get_datacard_yields.py [hunk 1]
```diff
@@ -298,6 +301,12 @@ def main():
         cat_name = get_cat_name_from_dc_name(dc_fname)
         all_rates_dict[cat_name] = rate_dict_sm
 
+    #printd(all_rates_dict)
+    #for k,v in all_rates_dict.items():
+        #print("\n",k)
+        #printd(v)
+    #exit()
+
```
Comment (bryates) [inline analysis/topEFT/get_datacard_yields.py:309]: Can we remove

File: analysis/topEFT/make_cr_and_sr_plots.py [hunk 2]
```diff
@@ -41,6 +41,9 @@
         "2lss_ee_CR_2j",
         "2lss_em_CR_2j",
         "2lss_mm_CR_2j",
+        #"2lss_ee_CR_3j",
+        #"2lss_em_CR_3j",
+        #"2lss_mm_CR_3j",
```
Comment (bryates) [inline analysis/topEFT/make_cr_and_sr_plots.py:46]: Remove?

File: topcoffea/modules/YieldTools.py [hunk 3]
```diff
@@ -415,32 +612,48 @@ def get_normalized_yield(self,hin_dict,year,proc,cat_dict,overflow_str,rwgt_pt=N
...
+    def get_yld_dict(self,hin_dict,year=None,njets=False,lepflav=False):
 
         # Check for options that do not make sense
         if lepflav and not self.is_split_by_lepflav(hin_dict):
             raise Exception("Error: Cannot split by lep flav if the input file is not split by lep flav.")
 
         # If we want to seperate by njets, don't use njets hist since njets are not in it's sparse axis
         hist_to_use = "njets"
-        if njets: hist_to_use = "ht"
+        #if njets: hist_to_use = "ht"
```
Comment (bryates) [inline topcoffea/modules/YieldTools.py:623]: Remove?

File: topcoffea/modules/YieldTools.py [hunk 4]
```diff
@@ -415,32 +612,48 @@ def get_normalized_yield(self,hin_dict,year,proc,cat_dict,overflow_str,rwgt_pt=N
...
             cat_dict[ch]["channel"] = ch
 
         # Find the yields
         yld_dict = {}
         proc_lst = self.get_cat_lables(hin_dict,"sample")
+        if "flipsUL17" not in proc_lst: proc_lst = proc_lst + ["flipsUL16","flipsUL16APV","flipsUL17","flipsUL18"] # Very bad workaround for _many_ reasons
+        print("proc_lst",proc_lst)
+        #for proc in proc_lst:
+        #    p = self.get_short_name(proc)
+        #    print("Name:",p,proc) # Print what name the sample has been matched to
```
Comment (bryates) [inline topcoffea/modules/YieldTools.py:642]: Remove?

**Review by bryates — APPROVED** [review]
Thanks for adding all of this. I don't have any comments, other than some commented code that we could probably remove.

File: analysis/topEFT/get_datacard_yields.py (hunk 1 above)
Comment (kmohrman) [reply analysis/topEFT/get_datacard_yields.py:305]: Removed

File: analysis/topEFT/make_cr_and_sr_plots.py (hunk 2 above)
Comment (kmohrman) [reply analysis/topEFT/make_cr_and_sr_plots.py]: Removed

File: topcoffea/modules/YieldTools.py (hunk 3 above)
Comment (kmohrman) [reply topcoffea/modules/YieldTools.py]: Removed

File: topcoffea/modules/YieldTools.py (hunk 4 above)
Comment (kmohrman) [reply topcoffea/modules/YieldTools.py]: Uncommented (I run with this uncommented more than commented, so probably better to uncomment it than to remove it)

**Pull request merged**